### PR TITLE
Add option to disable component registration

### PR DESCRIPTION
**Pull request by mvdnbrk**
Related to #71

If you are only using the blade directive with the `@svg` or `svg` helper there is no need to register all the components.

This PR adds a config option to disable the registration of the components.

### CODE REVIEW

**Comment by stidges** [comment]
I'd love to see this added in, as this is currently blocking me from upgrading to Laravel 8 (I'm still depending on the old blade-svg package at the moment). I'm not using the Blade components, and the load times on a basic page went up from 20ms to over 300ms per request.

**Comment by amberlampsio** [comment]
@driesvints

**Comment by driesvints** [comment]
Sorry still haven't gotten to this. I have a week vacation in two weeks so hoping to get to all of this then or even sooner.

File: tests/ComponentsTest.php [hunk 1]
```diff
@@ -33,6 +33,18 @@ public function components_are_registered_with_their_subdirectories()
         }
     }
 
+    /** @test */
+    public function componenets_are_not_registerd_when_disabled()
```
Comment (miclf) [inline tests/ComponentsTest.php]: Small typo here.

`componenets` ➡️ `components`

Comment (mvdnbrk) [reply tests/ComponentsTest.php]: Thanks, fixed.

**Comment by stidges** [comment]
For anyone blocked by this, here's a quick workaround. 

Leave your `'sets'` in the `blade-icons.php` config file empty, and register your icon set in your `AppServiceProvider` like so:

```php
<?php

use BladeUI\Icons\Factory as BladeIconsFactory;
use Illuminate\Contracts\View\Factory as ViewFactory;
use Illuminate\Foundation\Application;
use Illuminate\Support\ServiceProvider;

class AppServiceProvider extends ServiceProvider
{
    public function register()
    {
        // ...

        $this->callAfterResolving(ViewFactory::class, function ($view, Application $app) {
            $app->make(BladeIconsFactory::class)->add('default', [
                'path' => $app->resourcePath('svg'),
                'prefix' => 'icon',
                'class' => '',
            ]);
        });
    }
}
```

This works because in the `BladeIconsServiceProvider` it automatically registers the components after the `ViewFactory` is resolved. Placing this code in your `AppServiceProvider` will add the icon set *after* this automatic registration has run.

Here's the relevant code in the `BladeIconsServiceProvider` that we bypass this way: https://github.com/blade-ui-kit/blade-icons/blob/d6bb74cdf0420900d14c6bc2304e28d0463dd0a9/src/BladeIconsServiceProvider.php#L48-L50

**Comment by driesvints** [comment]
Heya, I've sent in an alternative to this one here: https://github.com/blade-ui-kit/blade-icons/pull/118

This will be in the next release. Thanks all!

**Pull request closed without merging**